src/App.tsx

```diff
@@ -15,47 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+const sortByType = (sortType: SortType, good1: string, good2: string) => {
+  switch (sortType) {
+    case SortType.ALPABET:
+      return good1.localeCompare(good2);
+
+    case SortType.LENGTH:
+      return good1.length - good2.length;
+
+    default:
+      return 0;
+  }
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => (
+    sortByType(sortType, good1, good2)
+  ));
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
```

Comment:
always put conditions in ternary on new lines
```suggestion
  return isReversed 
    ? visibleGoods.reverse() 
    : visibleGoods;
```